### PR TITLE
add learning rate to dart

### PR DESCRIPTION
**Pull request by wxchan**
refer to #126

### CODE REVIEW

File: src/boosting/dart.hpp [hunk 1]
```diff
@@ -106,23 +105,30 @@ class DART: public GBDT {
...
   }
   /*!
   * \brief normalize dropped trees
+  * NOTE: num_drop_tree(k), learning_rate(lr), shrinkage_rate_ = lr / (k + lr)
+  *       step 1: shrink tree to -1 -> drop tree
+  *       step 2: shrink tree to k * lr / (k + lr) - 1 from -1
+  *               -> normalize for valid data
+  *       step 3: shrink tree to k * lr / (k + lr) from k * lr / (k + lr) - 1
+  *               -> normalize for train data
+  *       end with tree weight = k * lr / (k + lr)
```
Comment (guolinke) [inline src/boosting/dart.hpp]: why multiply ```k``` ?

Comment (wxchan) [inline src/boosting/dart.hpp]: it's same as before. 
before:
weights of dropped trees: k / (1 + k)
weights of new trees:  1 / (1 + k)
now: 
change 1 to lr

Comment (wxchan) [inline src/boosting/dart.hpp]: sorry, wrong here. dropped tree should multiply k / (k + lr). wait a minute

Comment (wxchan) [inline src/boosting/dart.hpp]: It's updated. See latest code.
